src/main/java/org/deepsymmetry/beatlink/VirtualRekordbox.java

```diff
@@ -1380,1264 +1031,46 @@ private void deliverDeviceUpdate(final DeviceUpdate update) {
-            Collections.newSetFromMap(new ConcurrentHashMap<MediaDetailsListener, Boolean>());
-
-    /**
-     * <p>Adds the specified media details listener to receive detail responses whenever they come in.
-     * If {@code listener} is {@code null} or already present in the list
-     * of registered listeners, no exception is thrown and no action is performed.</p>
-     *
-     * <p>To reduce latency, device updates are delivered to listeners directly on the thread that is receiving them
-     * from the network, so if you want to interact with user interface objects in listener methods, you need to use
-     * <code><a href="http://docs.oracle.com/javase/8/docs/api/javax/swing/SwingUtilities.html#invokeLater-java.lang.Runnable-">javax.swing.SwingUtilities.invokeLater(Runnable)</a></code>
-     * to do so on the Event Dispatch Thread.</p>
-     *
-     * <p>Even if you are not interacting with user interface objects, any code in the listener method
-     * <em>must</em> finish quickly, or it will add latency for other listeners, and detail updates will back up.
-     * If you want to perform lengthy processing of any sort, do so on another thread.</p>
-     *
-     * @param listener the media details listener to add
+     * Holds the singleton instance of this class.
      */
-    public void addMediaDetailsListener(MediaDetailsListener listener) {
-        if (listener != null) {
-            detailsListeners.add(listener);
-        }
-    }
+    private static final VirtualRekordbox ourInstance = new VirtualRekordbox();
 
     /**
-     * Removes the specified media details listener so it no longer receives detail responses when they come in.
-     * If {@code listener} is {@code null} or not present
-     * in the list of registered listeners, no exception is thrown and no action is performed.
+     * Get the singleton instance of this class.
      *
-     * @param listener the media details listener to remove
+     * @return the only instance of this class which exists
      */
-    public void removeMediaDetailsListener(MediaDetailsListener listener) {
-        if (listener != null) {
-            detailsListeners.remove(listener);
-        }
+    public static VirtualRekordbox getInstance() {
+        return ourInstance;
     }
 
     /**
-     * Get the set of media details listeners that are currently registered.
-     *
-     * @return the currently registered details listeners
+     * Register any relevant listeners; private to prevent instantiation.
      */
-    public Set<MediaDetailsListener> getMediaDetailsListeners() {
-        // Make a copy so callers get an immutable snapshot of the current state.
-        return Collections.unmodifiableSet(new HashSet<MediaDetailsListener>(detailsListeners));
-    }
+    private VirtualRekordbox() {}
 
     /**
-     * Send a media details response to all registered listeners.
+     * We have received a packet from a device trying to claim a device number, see if we should defend it.
```

Comment:
Do we have any reason to believe the Opus sends these? Or do you think other rekordbox instances do?